src/actions/receive-profile.js

```diff
@@ -1407,7 +1407,11 @@ export function getProfilesFromRawUrl(
       // be 'from-file' and needs to be redirected to home page.
       dataSource = 'none';
     }
-    dispatch(setDataSource(dataSource));
+    const query = queryString.parse(location.search.substr(1), {
```

Comment:
Suggestion: Ok, coming back to this, I think this may be the wrong approach.

Right now when we load from-addon, we discard all URL state, which I believe is an error if we want to go this route. This current patch is fixing this one particular case, by doing some custom parsing of the query, and then acting on it. This is brittle as it only handles this one particular case. I believe we should do the correct thing of fully parsing the URL state from the location using `stateFromLocation`.

Perhaps what's needed is changing the line `dispatch(setDataSource(dataSource));` in this action to a setUrlState from a full `stateFromLocation` call.